src/fixate/reporting/csv.py

```diff
@@ -115,45 +115,56 @@ class CSVWriter:
+
+        self.exception_in_test = False
+        self.failed = False
+        self.chk_cnt = 0
+        self.csv_path = ""
+        self.test_module = None
+        self.start_time = None
+        self.current_test = None
+        self.data = fixate.config.get_config_dict()
+        self.data.update(fixate.config.get_plugin_data("plg_csv"))
+        self.exception = None
 
     def install(self):
-        self.csv_writer = ExcThread(
-            target=self._csv_write, args=(self.csv_queue,), name="csv-writer"
-        )
+        self.csv_writer = ExcThread(target=self._csv_write, name="csv-writer")
         self.csv_writer.start()
 
+        pub.subscribe(self.test_start, "Test_Start")
+        pub.subscribe(self.test_comparison, "Check")
+        pub.subscribe(self.test_exception, "Test_Exception")
+        pub.subscribe(self.test_complete, "Test_Complete")
+        pub.subscribe(self.sequence_update, "Sequence_Update")
+        pub.subscribe(self.sequence_complete, "Sequence_Complete")
+        pub.subscribe(self.user_wait_start, "UI_block_start")
+        pub.subscribe(self.user_wait_end, "UI_block_end")
+        pub.subscribe(self.driver_open, "driver_open")
+
     def uninstall(self):
+        pub.unsubscribe(self.test_start, "Test_Start")
+        pub.unsubscribe(self.test_comparison, "Check")
+        pub.unsubscribe(self.test_exception, "Test_Exception")
+        pub.unsubscribe(self.test_complete, "Test_Complete")
+        pub.unsubscribe(self.sequence_update, "Sequence_Update")
+        pub.unsubscribe(self.sequence_complete, "Sequence_Complete")
+        pub.unsubscribe(self.user_wait_start, "UI_block_start")
+        pub.unsubscribe(self.user_wait_end, "UI_block_end")
+
         if self.csv_writer:
             self.csv_queue.put(None)
             self.csv_writer.join()
         self.csv_writer = None
 
-    def _csv_write(self, cmd_q):
-        while True:
-            line = cmd_q.get()
-            if line is None:
-                break  # Command send to close csv_writer
-            try:
-                os.makedirs(os.path.dirname(self.reporting.csv_path))
-            except OSError as e:
-                pass
-            with open(self.reporting.csv_path, "a+", newline="") as f:
-                writer = csv.writer(f, quoting=csv.QUOTE_MINIMAL)
-                writer.writerow(line)
+    def is_alive(self):
+        if self.exception:
+            raise RuntimeError(
+                f"Exception in {self.csv_writer.name} thread"
+            ) from self.exception
 
-
-class CsvReporting:
-    def __init__(self):
-        self.exception_in_test = False
-        self.failed = False
-        self.chk_cnt = 0
-        self.csv_path = ""
-        self.test_module = None
-        self.start_time = None
-        self.current_test = None
-        self.data = fixate.config.get_config_dict()
-        self.data.update(fixate.config.get_plugin_data("plg_csv"))
+        if not self.csv_writer.is_alive():
```

Comment:
A minor nit-pick on the naming. If `is_alive` is never expected to return a value, I'd rename it to `ensure_alive`, `ensure_running` or similar. I'd expect an "is" function to return a bool